vector/src/main/java/im/vector/app/core/extensions/Context.kt

```diff
@@ -77,3 +83,31 @@ val Context.dataStoreProvider: (String) -> DataStore<Preferences> by dataStorePr
 fun Context.safeOpenOutputStream(uri: Uri): OutputStream? {
     return contentResolver.openOutputStream(uri, "wt")
 }
+
+/**
+ * Checks for an active connection to infer if the device is offline.
+ * This is useful for breaking down UnknownHost exceptions and should not be used to determine if a valid connection is present
+ *
+ * @return true if no active connection is found
+ */
+@Suppress("deprecation")
+@SuppressLint("NewApi") // false positive
+fun Context.inferNoConnectivity(buildMeta: BuildMeta): Boolean {
```

Comment:
Should we add a description about `buildMeta` parameter?

Reply:
will add some java doc :+1: